nomos-services/storage/src/backends/rocksdb.rs

```diff
@@ -0,0 +1,252 @@
+        }
+    }
+}
+
+impl<SerdeOp> core::fmt::Debug for RocksBackend<SerdeOp> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        format!("RocksBackend {{ rocks: {:?} }}", self.rocks).fmt(f)
+    }
+}
+
+#[async_trait]
+impl<SerdeOp: StorageSerde + Send + Sync + 'static> StorageBackend for RocksBackend<SerdeOp> {
+    type Settings = RocksBackendSettings;
+    type Error = rocksdb::Error;
+    type Transaction = Transaction;
+    type SerdeOperator = SerdeOp;
+
+    fn new(config: Self::Settings) -> Result<Self, Self::Error> {
+        let RocksBackendSettings {
+            db_path,
+            read_only,
+            column_family: cf,
+        } = config;
+
+        let db = match (read_only, cf) {
+            (true, None) => {
+                let mut opts = Options::default();
+                opts.create_if_missing(false);
+                DB::open_for_read_only(&opts, db_path, false)?
+            }
+            (true, Some(cf)) => {
+                let mut opts = Options::default();
+                opts.create_if_missing(false);
+                DB::open_cf_for_read_only(&opts, db_path, [cf], false)?
+            }
+            (false, None) => {
+                let mut opts = Options::default();
+                opts.create_if_missing(true);
+                opts.create_missing_column_families(true);
+                DB::open(&opts, db_path)?
+            }
+            (false, Some(cf)) => {
+                let mut opts = Options::default();
+                opts.create_if_missing(true);
+                opts.create_missing_column_families(true);
+                DB::open_cf(&opts, db_path, [cf])?
+            }
+        };
+
+        Ok(Self {
+            rocks: Arc::new(db),
+            _serde_op: Default::default(),
+        })
+    }
+
+    async fn store(&mut self, key: Bytes, value: Bytes) -> Result<(), Self::Error> {
+        self.rocks.put(key, value)
+    }
+
+    async fn load(&mut self, key: &[u8]) -> Result<Option<Bytes>, Self::Error> {
+        self.rocks.get(key).map(|opt| opt.map(|ivec| ivec.into()))
+    }
+
+    async fn remove(&mut self, key: &[u8]) -> Result<Option<Bytes>, Self::Error> {
+        self.load(key).await.and_then(|val| {
+            if val.is_some() {
+                self.rocks.delete(key).map(|_| val)
+            } else {
+                Ok(None)
+            }
+        })
+    }
+
+    async fn execute(
+        &mut self,
+        transaction: Self::Transaction,
+    ) -> Result<<Self::Transaction as StorageTransaction>::Result, Self::Error> {
+        Ok(transaction.execute())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::super::testing::NoStorageSerde;
+    use super::*;
+    use tempfile::TempDir;
+
+    #[tokio::test]
+    async fn test_store_load_remove(
+    ) -> Result<(), <RocksBackend<NoStorageSerde> as StorageBackend>::Error> {
+        let temp_path = TempDir::new().unwrap();
+        let sled_settings = RocksBackendSettings {
```

Comment:
```suggestion
        let rocks_settings = RocksBackendSettings {
```
also, in other tests